bundles/org.openhab.binding.icalendar/src/test/java/org/openhab/binding/icalendar/internal/logic/BiweeklyPresentableCalendarTest.java

```diff
@@ -542,4 +542,47 @@ public void testCommandTagCode() {
         assertNotNull(cmd7);
         assertEquals(QuantityType.class, cmd7.getClass());
     }
+
+    @SuppressWarnings("null")
+    @Test
+    public void testGetFilteredEventsBetween() {
+        Event[] expectedFilteredEvents1 = new Event[] {
+                new Event("Test Series in UTC", Instant.parse("2019-09-12T09:05:00Z"),
+                        Instant.parse("2019-09-12T09:10:00Z"), ""),
+                new Event("Test Event in UTC+2", Instant.parse("2019-09-14T08:00:00Z"),
+                        Instant.parse("2019-09-14T09:00:00Z"), "") };
+        List<Event> realFilteredEvents1 = calendar.getFilteredEventsBetween(Instant.parse("2019-09-12T06:00:00Z"),
+                Instant.parse("2019-09-15T06:00:00Z"), null, 3);
+        assertArrayEquals(expectedFilteredEvents1, realFilteredEvents1.toArray(new Event[0]));
```

Comment:
It's good practice to use one assert per test to see which test fails at the first view.

Reply:
I do not agree generally. The tests are built currently on each consecutively as we have more context after each assert - to create tests with one assert per test this is not practical, please look into `testIsEventPresent()` which shows what i mean. Doing this test in the same way is just consequent.

Reply:
It looked for me that each `expectedFilteredEvents*` is independet from each other and the asserts could be invoked in arbitrary order. It was only a suggestion.